raid/signup.py

```diff
@@ -70,12 +69,20 @@ async def handle_emoji(self, emoji, channel, msg, user, guild, member):
         if str(emoji) == SIGNUP_EMOJI:
             embed = msg.embeds[0]
             raid_worksheet = get_worksheet(embed.title)
-            discord_ids = col_values(raid_worksheet, 5)
+            
+            user_ids = col_values(raid_worksheet, 5)
             try:
-                delete_row(raid_worksheet, discord_ids.index(str(user)) + 1)
+                # Try to delete by the member id
+                delete_row(raid_worksheet, user_ids.index(str(member.id)) + 1)
             except ValueError:
-                # Treat deletion of an ID that doesn't exist as idempotent success
-                pass
+                # TODO: Can remove this try/except block after the migration is complete.
```

Comment:
I recommend doing the migration actively instead of lazily like this since there's no guarantee everyone will interact with the sheet. You can either throw it into the bot as a command or just run it offline

Reply:
I think doing a migration actively introduces more room for error. An active migration is more difficult as it requires converting currently active sheets where user IDs may already be changed, which is what this PR is looking to fix. This change being done per-raid sheet makes the migration pretty short, and the approach above allows us to have both member ID and discord ID in sheets live in harmony until the discord ID is fully obsolete.

Reply:
An easier way to put it: active migration is prone to the same bug that the code base is already vulnerable to

Reply:
I disagree - it would only be error prone in the way this is written. A properly written migration would be designed to take any existing potential inputs (corrupted or not) and transform them to the same desired output. I haven't dug around this too much but there should be a way to transform the existing data into the desired state for both cases and it's just a matter of classifying and executing the correct transform for each row.

Reply:
If there isn't then I guess it has to be done lazily unless you want to go sleuthing around for who has changed their name from what was initially put into the sheet, which sound awful lol

Reply:
Yeah, I know of a few cases where this is already the case (one of the BWL raid sheets), just didn't want to deal with manually fixing shit. Thanks for passing it over, will keep active migration in case something like this surfaces again